server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -379,9 +380,13 @@ public SnapshotInfo(SnapshotsInProgress.Entry entry) {
             Version.CURRENT,
             entry.startTime(),
             0L,
-            0,
-            0,
-            Collections.emptyList(),
+            entry.shards().size(),
+            Math.toIntExact(entry.shards().stream().filter(c -> c.getValue().state() == SnapshotsInProgress.ShardState.SUCCESS).count()),
```

Comment:
Might be nice to calculate both this count and the failures list in a single loop by using a static method + `Tuple` or so? Afaik this may run on a transport thread so it's worth saving the cycles, especially as we scale to larger and larger clusters.

Reply:
Yeah, I believe I can do that with a `teeing` collector.

Reply:
Actually, we can't, because we were calling another constructor and couldn't declare variables. I went ahead with removing of the constructor call and using a simple loop for calculating both fields.